### ISSUE TITLE
Implement search

### ISSUE THREAD

**Issue by richvdh** [matrix-org/matrix.org#19]
From https://matrix.org/jira/browse/ORG-57

[we have a search box but it only searches the wordpress blog afaict]

**Comment by benparsons** [comment]
Global search + nav to be revamped, this could be a part of it. I'd like to satisfy navigation without needing to add Google Search, but to be considered.

**Comment by aaronraimist** [comment]
Yeah the search really needs to be improved, it is pretty unusable right now. Replacing it with either Google or DuckDuckGo or something else would be much appreciated.

**Comment by aaronraimist** [comment]
I can't seem to fix this because the search icon in my local dev build doesn't work at all but I think it should be as simple as sticking this https://stackoverflow.com/questions/7259209/simple-site-search-with-duckduckgo in here https://github.com/matrix-org/matrix.org/blob/7403ff54ac63ed07092f9c10bf4d44dd6e405122/jekyll/_includes/nav.html#L23

**Comment by thibaultamartin** [comment]
Instead of a Google search box that will likely be received negatively by the audience, we should rely on Zola's "built in" search: https://www.getzola.org/documentation/content/search/

**Comment by thibaultamartin** [comment]
I've had some positive experience with [Pagefind](https://pagefind.app/) on static websites. It's a bunch of javascript that provides two webcomponents to add to our pages.

At build time, pagefind indexes all the content and generates a dictionary(ish?) that is called by the webcomponents later.

One thing to bear in mind is that [Cloudflare Pages limits the files size to 25MiB](https://developers.cloudflare.com/pages/platform/limits/#file-size), so the index can't be larger than that.
